Added optional flag to turn off display in POVRaytracer render

joshichaitanya3 opened the pull request:
When rendering `POVRaytracer` objects in a loop, one might want to turn graphic display off. I added an optional `display=true` flag to the `render` method that achieves this, while not affecting the default behavior. When `display=false`, an optional flag of `-D` is added to the `povray` system call, which turns off the graphic display.

I am not sure how to add an automated test for this, but here is a MWE that generates `test.pov` and `test.png` without opening up the graphic display:

`import meshtools`
`import plot`
`import povray`

`var m = LineMesh(fn (t) [t,0,0], -1..1:0.2)`

`var g = plotmesh(m)`
`var pov = POVRaytracer(g)`
`pov.render("test.pov", display=false)`

Removing the `display` flag or setting it to `true` should result in the default behavior. Tested on Ubuntu 20.04 and on WSL-Ubuntu on Windows.

ConduitDan reviewed (approved):
This looks good to me
I don't know of a good way of testing this change. Perhaps there is a system call that can check if there is a povray window open?

softmattertheory reviewed (commented):
Looks good to me. Could we document this in docs/povray.md?

joshichaitanya3 commented:
Thanks for the feedback! I have now added documentation to [docs/povray.md](https://github.com/joshichaitanya3/morpho/blob/povray-display-option/morpho5/docs/povray.md). 

I also updated the `render` function so that the .png file too is not opened if `display=false`.

I noticed that the `render` function opens the .png file calling the `open` command, but that doesn't work on Linux (I believe `xdg-open` should work in most cases for Linux). The function would need to check what OS it's running on and change that command accordingly, but I don't know how to do that.

I also noticed that the .md file needs to end in an empty line for the documentation to be displayed properly in the CLI. If there is not empty line, then the last line is not displayed unless we hit Enter again.
![Inkedweird_help_display_LI](https://user-images.githubusercontent.com/16670570/140660663-a415c241-e693-4a5d-89e3-f7d9fe79bc05.jpg)
I am guessing the blank line makes things easier and is advised to add it to every documentation?

softmattertheory commented:
Indeed replacing open with something more cross platform is a good idea (a feature for 0.6 I think!) In the meantime, symlink xdg-open to open.
